src/main/java/com/arpnetworking/metrics/impl/BaseFileSink.java

```diff
@@ -414,5 +453,7 @@ protected void validate(final List<String> failures) {
         private static final Boolean DEFAULT_PRUDENT = Boolean.FALSE;
         private static final Boolean DEFAULT_DROP_WHEN_QUEUE_FULL = Boolean.FALSE;
         private static final Integer DEFAULT_MAX_QUEUE_SIZE = Integer.valueOf(500);
+        private static final String DEFAULT_MAX_FILE_SIZE = "100MB";
+        private static final String DEFAULT_TOTAL_SIZE_CAP = "20GB";
```

Comment:
No default max size can easily return a time based rolling policy only.

totalSizeCap = maxHistory * maxFileSize  only makes sense when the compression is set to false.

Testing enforcement of max size is difficult in the current release of logback due to code in the archive remover that guards the most recent two periods of time (depending on your time based rolling schedule).  This code has been removed in the current master and I was able to write a good test.